packages/blade/src/components/Radio/useRadio.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import { useControllableState } from '../../hooks/useControllable';
+import { useId } from '../../hooks/useId';
+import { getPlatformType, makeAccessible } from '../../utils';
+import type { RadioProps } from './Radio';
+
+type UseRadioProps = Pick<
+  RadioProps,
+  'isChecked' | 'defaultChecked' | 'isDisabled' | 'isRequired' | 'onChange' | 'name' | 'value'
+> & { hasError?: boolean };
+
+const useRadio = ({
+  isChecked,
+  defaultChecked,
+  isDisabled,
+  isRequired,
+  hasError,
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
+  if (isChecked && defaultChecked) {
+    throw new Error(
+      `[Blade useRadio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
+    );
+  }
+
+  const [radioState, setRadioStateChange] = useControllableState({
+    value: isChecked,
+    defaultValue: defaultChecked ?? false,
+  });
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement> | GestureResponderEvent) => {
+    if (isDisabled) {
+      event.stopPropagation();
+      event.preventDefault();
```

Comment:
These might not be available on react native. Can you check and ensure this doesn't break on react native?

Reply:
Nope doesn't seem to break, This code is also present on Checkbox. I think React's synthetic event system mocks these on native.